extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsStorageAuthentication.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.storage.hdfs;
+
+
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.metamx.common.ISE;
+import com.metamx.common.lifecycle.LifecycleStart;
+import com.metamx.common.lifecycle.LifecycleStop;
+import com.metamx.common.logger.Logger;
+import io.druid.guice.ManageLifecycle;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import java.io.IOException;
+import java.util.Properties;
+
+@ManageLifecycle
+public class HdfsStorageAuthentication
+{
+  private static final Logger log = new Logger(HdfsStorageAuthentication.class);
+  private final Properties props;
+  private final Configuration conf;
+
+  @Inject
+  public HdfsStorageAuthentication(Properties properties, Configuration conf)
+  {
+    this.props = properties;
+    this.conf = conf;
+  }
+
+  /**
+   * Dose authenticate against a secured hadoop cluster
+   * In case of any bug fix make sure to fix the code in JobHelper#authenticate as well.
+   */
+  @LifecycleStart
+  public void authenticate()
+  {
+    final String principal = props.getProperty("druid.hdfs.security.kerberos.principal");
+    final String keytab = props.getProperty("druid.hdfs.security.kerberos.keytab");
```

Comment:
I mean, using a class bound to properties, and injecting that class, instead of injecting directly here.

Im not totally familiar with how the kerberos auth for hadoop works, but if there is secret matter not suitable for storage in properties we should extend the password provider some more (or similar) to work with secret matters that might not be strings.